scalafix-core/src/main/scala/scalafix/v1/SemanticDocument.scala

```diff
@@ -20,6 +20,8 @@ final class SemanticDocument private[scalafix] (
     tree.tokens
   def input: Input =
     internal.doc.internal.input
+  def uri: String =
```

Comment:
Thanks for this contribution! Would it be better to return the original Scala path instead of a relative URI string? `Paths.get(input.syntax)` returns the absolute path to the source file. It's quite tricky to properly absolutize the `uri` string because it's actually URI encoded so you need to for example handle spaces correctly.

Reply:
Also, `input.syntax` is available for both semantic and syntactic rules while the `uri` field is only available for semantic rules.